template-manager-core/src/main/java/com/sequenceiq/cloudbreak/template/TemplatePreparationObject.java

```diff
@@ -53,6 +53,8 @@
 
```

Comment:
**question:** What is this `TemplatePreparationObject` for? It looks like just a plain old value object.

I see it's in the `template` namespace, but I don't actually know what that means in the broader context of CB.

Reply:
@brycederriso read more here https://cloudera.atlassian.net/wiki/spaces/ENG/pages/730268997/TRAINING+-+Cloudera+Manager+template+generation

Reply:
Yes, it is am immutable POJO that contains all the cluster & service details needed for dynamic template generation. As explained in the wiki that Richie linked, the `TemplatePreparationObject` is fed into two kinds of template engines: Handlebars (used for cloud storage location definitions & dynamic parameter placeholders within CM cluster templates) and the CB logic generating CM cluster templates.

Reply:
Thanks!